src/main.c

```diff
@@ -569,7 +570,11 @@ char * generateRequest(void *bigkey, char *date) {
   char *dump = malloc(513);
   char *result = malloc(1024); // base64-encoded code is 680 bytes
   
-  memset(iv, 0x42, mcrypt_enc_get_iv_size(blowfish));
+  // use a fresh, random iv for each request
+  if ((ssize_t) mcrypt_enc_get_iv_size(blowfish) != getrandom(iv, mcrypt_enc_get_iv_size(blowfish), GRND_NONBLOCK)) {
+    PERROR("%s", "initializing random rv");
```

Comment:
is it a typo, i.e. you probably meant "iv"?

Reply:
yes, typo, thanks, fixed